slang-kotlin/src/main/java/com/sonarsource/slang/kotlin/KotlinTreeVisitor.java

```diff
@@ -187,13 +189,20 @@ private Tree createElement(@Nullable PsiElement element) {
       }
       return new MatchCaseTreeImpl(getTreeMetaData(whenElement), conditions, body);
     } else if (element instanceof KtStringTemplateExpression) {
-      KtStringTemplateEntry[] entries = ((KtStringTemplateExpression) element).getEntries();
-      if (entries.length == 1 && entries[0] instanceof KtLiteralStringTemplateEntry) {
-        // Non-template strings, ie. not in the form "string ${1 + 1}"
-        return new LiteralTreeImpl(metaData, '\"' + entries[0].getText() + '\"');
+      boolean complexString = Arrays.stream(((KtStringTemplateExpression) element).getEntries())
+        .anyMatch(KtStringTemplateEntryWithExpression.class::isInstance);
+      if (complexString) {
+        return new NativeTreeImpl(metaData, new KotlinNativeKind(element), list(Arrays.stream(((KtStringTemplateExpression) element).getEntries())));
```

Comment:
Maybe we can factorize a bit by creating a variable for `((KtStringTemplateExpression) element).getEntries()`. In fact, replacing this expression with `element.getChildren()` doesn't seem to break the tests...

Reply:
True. Actually I couldn't find any case where `element.getChildren()` would result in a different Tree than using `getEntries()`. And if such a case happens, it makes sense to use the children anyway in order to have different `NativeTree` in the end